moveit_planners/ompl/ompl_interface/src/planning_context_manager.cpp

```diff
@@ -114,6 +116,9 @@ static ompl::base::PlannerPtr allocatePlanner(const ob::SpaceInformationPtr& si,
   if (!new_name.empty())
     planner->setName(new_name);
   planner->params().setParams(spec.config_, true);
+  // Some planners (AnytimePathShortening) require problem and goal to be initialized before setup
```

Comment:
I added this because `AnytimePathShortening::setup()` would cause a segfault if `ProblemDefinition` was undefined and would throw an exception without a `GoalState` set. Internally, this is only used inside [ompl::tools::SelfConfig::getDefaultPlanner()](https://ompl.kavrakilab.org/classompl_1_1tools_1_1SelfConfig.html#ae9f747585a0c3fdc50667d9f8a39095d) for initializing the actual planner.

Reply:
It seems less than ideal that the MoveIt code needs to know about OMPL internals. Maybe AnytimePathShortening needs to be changed so that it can behave in the same way as all other planners.

Reply:
I agree, but in general I would understand that some planners require a valid problem definition for running `setup()`. In this case I think it's fine to keep these changes.

Reply:
This would definitely make it a lot easier to parameterize the planner type we want to use. We just still need to set the problem definition or otherwise `AnytimePathShortening::setup()` produces a segfault when calling `getDefaultPlanner()`. Anyway, I think it would make sense to add a custom allocator function that uses your factory so this wouldn't be an issue.

Reply:
I have pushed the changes above to the OMPL repo.

MoveIt calls Planner::setProblemDefinition at some point. The AnytimePathShortening class overrides this method to propagate the pdef to the different planner instances it contains, so my changes make the behavior of AnytimePathShortening consistent with the other planners.

Reply:
Since OMPL has been updated, can you refactor the MoveIt code to now use the AnytimePathShortening::createPlanner<Planner> factory method?